Library/Homebrew/test/cli_parser_spec.rb

```diff
@@ -0,0 +1,62 @@
+require_relative "../cli_parser"
+
+describe HomebrewCli::Parser do
+  describe "test boolean flags" do
+    it "with a valid flag and valid arg" do
+      parser = HomebrewCli::Parser.new do
+        option :verbose, description: "Flag for verbosity", type: :bool
```

Comment:
Could we define an RSpec `subject` here with a single parser created with a bunch of different options? Would make a big difference to the length of each test, which is a big win IMO.